### PR TITLE
feat: automatically generated instanceId

### PR DESCRIPTION
**Pull request by nunogois**
https://linear.app/unleash/issue/2-2393/make-instance-id-autogenned-c

This drops the `InstanceTag` public property in `UnleashSettings` in favor of an internal automatically generated `InstanceId` property. Any reference of `InstanceTag` is now `InstanceId`.

File names no longer include `InstanceTag` and instead we're relying on [gracefully handling file locks](https://github.com/Unleash/unleash-client-dotnet/blob/main/tests/Unleash.Tests/IOTests.cs).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -425,24 +424,7 @@ By default unleash-client fetches the feature toggles from unleash-server every
...
-    // Set an instance tag for consistent backup file naming
-    InstanceTag = "CustomInstanceTag",
-    UnleashContextProvider = new AspNetContextProvider(),
-    CustomHttpHeaders = new Dictionary<string, string>()
-    {
-        {"Authorization", "API token" }
-    }
-};
-```
+The backup file name will follow this pattern: `{fileNameWithoutExtension}-{AppName}-{SdkVersion}.{extension}`.
```
Comment (sighphyre) [inline README.md:427]: Probably worth mentioning that we took this out and why

**Comment by sighphyre** [comment]
How does this play with this:

> https://github.com/Unleash/unleash-client-dotnet/blob/93f4d88a1058581960752d754954dee665a16435/src/Unleash/Scheduling/FetchFeatureTogglesTask.cs#L92

If I'm reading that right this is still spitting out an error onto the error events. I think it shouldn't do that anymore WDYT?

**Review by sighphyre — APPROVED** [review]
I think we should change the error event but I don't think it's dangerous if you disagree so I'm gonna put an LGTM on it

**Comment by nunogois** [comment]
> How does this play with this:
> 
> > https://github.com/Unleash/unleash-client-dotnet/blob/93f4d88a1058581960752d754954dee665a16435/src/Unleash/Scheduling/FetchFeatureTogglesTask.cs#L92
> 
> If I'm reading that right this is still spitting out an error onto the error events. I think it shouldn't do that anymore WDYT?

Yep, addressed in https://github.com/Unleash/unleash-client-dotnet/pull/226/commits/78cf8c291e8d99d03bbaf9b80c69a30f74f714a8

**Pull request merged**